### PR TITLE
[86bxzn2xk][data-table] changed sort icon behaviour

### PR DESCRIPTION
**Pull request by ilyabrower**
## Motivation and Context
Changed sort icon behaviour

## How has this been tested?
Manually

## Screenshots (if appropriate):

https://github.com/semrush/intergalactic/assets/8787452/904daa48-fe6d-4078-90dc-2dd6681c7f08

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [X] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [X] Nice improve.

## Checklist:

- [X] My code follows the code style of this project.
- [X] I have updated the documentation accordingly or it's not required.
- [X] Unit tests are not broken.
- [X] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

### CODE REVIEW

File: semcore/data-table/src/Body.tsx [hunk 1]
```diff
@@ -96,8 +96,6 @@ class Body extends Component<AsProps, State> {
           borderRight: lastColumn?.borderRight,
           style: {
             width: vars.length === 1 ? vars[0] : `calc(${vars.join(' + ')})`,
-            minWidth: column?.props?.wMin,
-            maxWidth: column?.props?.wMax,
```
Comment (msereniti) [inline semcore/data-table/src/Body.tsx:100]: Proposing:
```diff
- maxWidth: column?.props?.wMax,
+maxWidth: column?.props?.wMax && sortIconWidth ? `calc(${sortIconWidth}px + ${column?.props?.wMax})` : column?.props?.wMax 
```

File: semcore/data-table/src/Head.tsx [hunk 2]
```diff
@@ -48,10 +52,86 @@ class Head extends Component<AsProps> {
...
-    setRef(props.ref, ref);
-    if (props.forwardRef) {
-      setRef(props.forwardRef, ref);
+  refSort = (active: boolean) => (ref: HTMLElement | null) => {
+    if (ref) {
+      this.sortWrapperRefs.set(ref, active);
+    }
+  };
+
+  refColumn = (column: Column) => (ref: HTMLElement | null) => {
```
Comment (msereniti) [inline semcore/data-table/src/Head.tsx:61]: Let's rename `refSort` and old `refColumn` to more correct `makeSortRefHandler` and `makeColumnRefHandler`

File: semcore/data-table/src/Head.tsx [hunk 3]
```diff
@@ -48,10 +52,86 @@ class Head extends Component<AsProps> {
     }
   };
 
-  refColumn = (props: Column['props']) => (ref: HTMLElement) => {
-    setRef(props.ref, ref);
-    if (props.forwardRef) {
-      setRef(props.forwardRef, ref);
+  refSort = (active: boolean) => (ref: HTMLElement | null) => {
```
Comment (msereniti) [inline semcore/data-table/src/Head.tsx:55]: It it ok that function doesn't consider what column it relates to? Afaik we may have more than one sortable column

File: semcore/data-table/src/Head.tsx [hunk 4]
```diff
@@ -48,10 +52,86 @@ class Head extends Component<AsProps> {
...
+
+      styles.forEach((key) => {
+        clonedColumn.style.setProperty(
+          key,
+          computedStyle.getPropertyValue(key),
+          computedStyle.getPropertyPriority(key),
+        );
+      });
+
+      clonedColumn.style.setProperty('visibility', 'hidden', 'important');
```
Comment (msereniti) [inline semcore/data-table/src/Head.tsx:112]: Let's put it before `styles.forEach((key) => {`

**Review by j-mnizhek — APPROVED** [review]

File: semcore/data-table/src/Head.tsx (hunk 4 above)
Comment (ilyabrower) [reply semcore/data-table/src/Head.tsx:112]: fixed

File: semcore/data-table/CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,12 @@
 
 CHANGELOG.md standards are inspired by [keepachangelog.com](https://keepachangelog.com/en/1.0.0/).
 
+## [4.35.0] - 2024-04-20
+
+### Changed
+
+- Sort icon behaviour.
```
[the remark this answers is not in the document]
Comment (ilyabrower) [reply semcore/data-table/CHANGELOG.md:9]: changed

File: semcore/data-table/src/Body.tsx (hunk 1 above)
Comment (ilyabrower) [reply semcore/data-table/src/Body.tsx:100]: yes, added it

File: semcore/data-table/src/Head.tsx (hunk 2 above)
Comment (ilyabrower) [reply semcore/data-table/src/Head.tsx:61]: renamed

File: semcore/data-table/src/Head.tsx (hunk 3 above)
Comment (ilyabrower) [reply semcore/data-table/src/Head.tsx:55]: As I see - no, we could have only one sortable column in a time

**Review by msereniti — APPROVED** [review]

**Pull request merged**